src/main/java/mate/academy/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -23,13 +25,18 @@ public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
     private final BookMapper bookMapper;
     private final BookSpecificationBuilder bookSpecificationBuilder;
+    private final CategoryService categoryService;
 
     @Override
     public BookDto save(BookRequestDto createBookRequestDto) {
         if (existsByIsbn(IsbnFormatter.format(createBookRequestDto.getIsbn()))) {
             throw new NotUniqueValueException("ISBN must be unique");
         }
         Book book = bookMapper.toModel(createBookRequestDto);
+        if (doesCategoryExist(createBookRequestDto)) {
+            book.setCategories(categoryService
+                    .getCategories(createBookRequestDto.getCategoriesId()));

```

Comment:
if you pass a non existing id among the existing ones it will be ignored, we should throw exception in that case, try to compare the sizes of collections that we received in the request and got from the DB